modules/tinymce/src/core/main/ts/api/Mode.ts

```diff
@@ -1,16 +1,38 @@
-import { Cell, Fun } from '@ephox/katamari';
+import { Cell, Fun, Obj, Type } from '@ephox/katamari';
 
 import { registerMode, setMode } from '../mode/Mode';
 import { isReadOnly, registerReadOnlyContentFilters, registerReadOnlySelectionBlockers } from '../mode/Readonly';
 import Editor from './Editor';
 
+export type ReadonlyProperty = Readonly<'uiEnabled' | 'selectionEnabled'>;
+
+export type EditorReadOnlyType = boolean | {
+  [K in ReadonlyProperty]?: boolean;
+};
+
 /**
  * TinyMCE Editor Mode API.
  *
  * @class tinymce.EditorMode
  */
 
 export interface EditorMode {
+  /**
+   * Checks if the editor content can be selected.
+   *
+   * @method selectionIsReadOnly
+   * @return {boolean} true if the editor content area allows selection.
+   */
+  isSelectionEnabled: () => boolean;
+
+  /**
+   * Checks if the editor user interface is in a readonly state.
+   *
+   * @method uiIsReadOnly
+   * @return {boolean} true if the editor user interface is enabled.
+   */
+  isUiEnabled: () => boolean;
+
   /**
    * Checks if the editor is in a readonly state.
```

Comment:
I think the meaning of this has changed a bit now

```suggestion
   * Checks if the editor content is in a readonly state.
```